File: sprint8/By languages/C#/lesson5code1.cs
int N = s.Length;
int[] π = new int[N];

for (int i = 1; i < N; i++)
{
    string substring = s.Substring(0, i);
    for (int k = i - 1; k >= 0; k--)
    {
        string prefix = substring.Substring(0, k);
        string suffix = substring.Substring(i - k, k);
        if (prefix == suffix)
        {
            π[i - 1] = k;
            break;
        }
    }
}

File: sprint8/By languages/Java/lesson5code1.java
int N = s.length();
int[] π = new int[N];

for (int i = 1; i < N; i++) {
    String substring = s.substring(0, i);
    for (int k = i - 1; k >= 0; k--) {
        String prefix = substring.substring(0, k);
        String suffix = substring.substring(i - k, i);
        if (prefix.equals(suffix)) {
            π[i-1] = k;
            break;
        }
    }
}
